### ISSUE TITLE
Unable to upload via csv or paste raw csv

### ISSUE THREAD

**Issue by daygon2007** [modxcms/Redirector#58]
I am unable to upload via csv or copy/paste raw csv. Not sure if it's my formatting or not but it keeps saying it's found the right amount of redirects but they all fail. I've tried different combinations and have not been able to get any of them to work.

Below are combinations I've tried using a real example:

blog/dont-let-grinch-steal-christmas/;[[~1447]]
blog/dont-let-grinch-steal-christmas/;[[~1447]];
blog/dont-let-grinch-steal-christmas/;[[~1447]];;

"blog/dont-let-grinch-steal-christmas/;[[~1447]]"
"blog/dont-let-grinch-steal-christmas/;[[~1447]];"
"blog/dont-let-grinch-steal-christmas/;[[~1447]];;"

blog/dont-let-grinch-steal-christmas/;[[~1447]],
blog/dont-let-grinch-steal-christmas/;[[~1447]];,
blog/dont-let-grinch-steal-christmas/;[[~1447]];;,

"blog/dont-let-grinch-steal-christmas/;[[~1447]]",
"blog/dont-let-grinch-steal-christmas/;[[~1447]];",
"blog/dont-let-grinch-steal-christmas/;[[~1447]];;",

"blog/dont-let-grinch-steal-christmas/;[[~1447]],"
"blog/dont-let-grinch-steal-christmas/;[[~1447]];,"
"blog/dont-let-grinch-steal-christmas/;[[~1447]];;,"

This is what I get when I do a print_r on for the array:
(Array
(
    [0] =&gt; Array
        (
            [0] =&gt; blog/dont-let-grinch-steal-christmas/
            [1] =&gt; [[~1447]]
            [2] =&gt; 
        )

)
{"success":true,"message":"Total 1 records found. Imported: 0 successful and 1 failed!","total":0,"data":[],"object":[]})

![screen shot 2015-09-25 at 11 08 50 am](https://cloud.githubusercontent.com/assets/12433369/10105722/db7ad550-6375-11e5-8929-42428e0cae45.png)

Any help or guidance would be awesome.

Bulk upload would be really awesome because I have over 700 redirects to put in place.

**Comment by daygon2007** [comment]
UPDATE:
So i performed a fresh install with no additional "extras". The script still does not work, however, if I use commas instead of semi-colons it will successfully insert the information but it only inserts the information in the pattern field, id does not put all of the info in the right place, as soon as I use a semi-colon again it will fail every single time.

I have checked log files to see if i can see where the failure is happening and I have checked the console. I can;t see to find where my issue is.

**Comment by nickff** [comment]
@daygon2007 Did you ever figure this one out? Let me know!

**Comment by daygon2007** [comment]
@nickff no unfortunately I had to do all 700+ imports 1 by 1 :( 👎

**Comment by getandyjones** [comment]
I cannot import CSV or paste raw data either - any update on this?

**Comment by labr1005** [comment]
Yeah, I just ran into this, too. The validation for target (and pattern) currently only checks for a valid resource uri: https://github.com/modxcms/Redirector/blob/develop/core/components/redirector/processors/mgr/import/import.class.php#L111

Regular expressions and MODX tags will always fail for target and pattern.

**Comment by daygon2007** [comment]
@labr1005 Thanks for catching that. That helped me out a lot because we're migrating part of our site to another install on a sub domain, so what i did was create a snippet to setup a php 301 redirect...

Here's the snippet named 301_redirect
```
<?php
header("HTTP/1.1 301 Moved Permanently"); 
header("Location: http://www.example.com/$uri");
```

And I replaced all of the code in my template with this `[[!301_redirect? &uri=`[[*uri]]`]]` because for our case it was going to be a direct 1-to-1 redirect to the new subdomain and the uri schema was going to be the exact same.
